### PR TITLE
cp-18: Add Protected Route Wrapper Component

### PR DESCRIPTION
**Pull request by OlenaYatseniuk**
Changes:
- added ProtectedRoute wrapper component ( for now I used mockUserData function only because we don't have user data in the redux state yet, so in the future, I will change it to get data from the state );
- removed root route content from App.tsx file;
- added root route page to move root route content to it;
- added protected route wrapper component to index.tsx file to wrap the Root component by it.

P.S. I don't attach a screenshot, because my task doesn't have any visual part.

### CODE REVIEW

**Comment by liza-veis** [comment]
You need to pull latest changes from development to your branch

File: frontend/src/libs/components/protected-route/protected-route.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+const mockUserData = (isUserExist: boolean): boolean => {
+  return isUserExist;
+};
+
+const ProtectedRoute: React.FC<Properties> = ({
+  component: Component,
+  ...rest
+}) => {
+  const hasUser = mockUserData(false);
```
Comment (v0ldemar01) [inline frontend/src/libs/components/protected-route/protected-route.tsx]:
```suggestion
  const { user } = useSelector(state => ({
    user: null
  }));

  const hasUser = Boolean(user);
  ```

**Review by v0ldemar01 — CHANGES_REQUESTED** [review]

File: frontend/src/libs/components/protected-route/protected-route.tsx (hunk 1 above)
Comment (OlenaYatseniuk) [reply frontend/src/libs/components/protected-route/protected-route.tsx]: Thanks. I've changed it.

File: frontend/src/libs/components/protected-route/protected-route.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import { Navigate } from 'react-router-dom';
+
+import { AppRoute } from '#libs/enums/enums.js';
+
+type Properties = {
+  component: ComponentType;
+};
+
+const ProtectedRoute: React.FC<Properties> = ({
+  component: Component,
```
Comment (v0ldemar01) [inline frontend/src/libs/components/protected-route/protected-route.tsx]: Let's change into children for component prop. And also add redirectPath with the default value AppRoute.SIGN_IN

Comment (v0ldemar01) [reply frontend/src/libs/components/protected-route/protected-route.tsx]: Sorry, this comment hasn't put on the preview request change)

Comment (OlenaYatseniuk) [reply frontend/src/libs/components/protected-route/protected-route.tsx]: It's OK. I've just done it.

**Review by v0ldemar01 — DISMISSED** [review]

**Review by fshabanov — DISMISSED** [review]

**Review by v0ldemar01 — DISMISSED** [review]

**Comment by OlenaYatseniuk** [comment]
I have updated my branch regarding the development branch and have solved one conflict. Could you, please, approve it again?

**Review by fshabanov — DISMISSED** [review]

File: frontend/src/libs/components/protected-route/protected-route.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import { type ReactNode } from 'react';
+import { useSelector } from 'react-redux';
+import { Navigate } from 'react-router-dom';
+
+import { AppRoute } from '#libs/enums/enums.js';
+
+type Properties = {
+  children: ReactNode;
+  redirectPath?: keyof typeof AppRoute;
```
Comment (liza-veis) [inline frontend/src/libs/components/protected-route/protected-route.tsx]:
```suggestion
  redirectPath?: ValueOf<typeof AppRoute>;
```

File: frontend/src/libs/components/protected-route/protected-route.tsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+};
+
+const ProtectedRoute: React.FC<Properties> = ({
+  children,
+  redirectPath = AppRoute.SIGN_IN,
+}) => {
+  const { user } = useSelector(() => ({
```
Comment (liza-veis) [inline frontend/src/libs/components/protected-route/protected-route.tsx:18]: Add `TODO: message` comment here)

**Review by liza-veis — CHANGES_REQUESTED** [review]

File: frontend/src/libs/components/protected-route/protected-route.tsx (hunk 3 above)
Comment (OlenaYatseniuk) [reply frontend/src/libs/components/protected-route/protected-route.tsx]: Have done it!

File: frontend/src/libs/components/protected-route/protected-route.tsx (hunk 4 above)
Comment (OlenaYatseniuk) [reply frontend/src/libs/components/protected-route/protected-route.tsx:18]: Have done it!

**Review by liza-veis — APPROVED** [review]

**Pull request merged**